proto/buf/alpha/wasmplugin/v1/wasmplugin.proto

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.wasmplugin.v1;
+
+// Meta describes buf specific extensions for a wasm plugin.  A metadata may
+// be encoded in a custom WASM section named ".bufplugin", see
+// (https://webassembly.github.io/spec/core/binary/modules.html#binary-customsec)
+// for more info.
+message Meta {
+  // When no ABI is provided, buf will make a best effort guess based on
+  // the functions the wasm module exports.
+  WasmABI wasm_abi = 1;
+
+  // The arguments that should be passed when running this plugin. Useful
+  // for interpreted languages where the main wasm bundle is only the
+  // interpreter.
+  repeated string args = 2;
+
+  // Files that should be made available to the WASI fs when running this
+  // plugin. Useful for interpreted languages where the main wasm bundle
+  // is only the interpreter.
+  repeated File files = 3;
+}
+
+// WASM_ABI specifies the abi this module expects buf to provide.
+enum WasmABI {
+  WASM_ABI_UNSPECIFIED = 0;
+
+  // Plugins compiled against
+  // https://github.com/WebAssembly/WASI/releases/tag/snapshot-01.
+  WASM_ABI_WASI_SNAPSHOT_PREVIEW1 = 1;
```

Comment:
Not sure if we want to use something more like `WASM_ABI_WASI_P1` to more closely follow https://github.com/golang/go/issues/58141.

Reply:
The wazero/experimental module is still `wasi_snapshot_preview1` so I am hesitant to change this. Think we should?

Reply:
Nah we can leave it.